source/serializer.c

```diff
@@ -214,8 +215,15 @@ int aws_iot_st_msg_deserialize_from_cursor(
                 goto cleanup;
             }
             aws_byte_cursor_advance(cursor, length);
+            payload_check = 1;
         }
     }
+    if (payload_check == 0) {
+        message->payload.allocator = NULL;
```

Comment:
Logic here may be OK to just zero memory on the entire struct.

Reply:
done